### PR TITLE
Implement `toMatchDiffSnapshot` jest matcher.

### PR DESCRIPTION
**Pull request by umidbekkarimov**
See #10

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -64,4 +65,12 @@ function diffReactComponents(valueA: any, valueB: any, options: Options) {
   });
 }
 
+function toMatchDiffSnapshot(valueA: any, valueB: any, options: Options) {
```
Comment (thymikee) [inline src/index.js]: `options?: Options`

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  expand?: boolean,
+  colors?: boolean,
+  contextLines?: number,
+};
+
+declare namespace jest {
+  interface Matchers {
+    toMatchDiffSnapshot(valueB: any, options?: DiffOptions): boolean
+  }
+}
```
Comment (thymikee) [inline index.d.ts]: newline pls

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/// <reference types="jest"/>
```
Comment (thymikee) [inline index.d.ts]: Is this valid in TS land?

File: __tests__/toMatchDiffSnapshot.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    very
+    long
+    script
+`;
+
+beforeAll(() => {
+  expect.extend({ toMatchDiffSnapshot: snapshotDiff.toMatchDiffSnapshot });
+});
+
+test('foo', () => {
```
Comment (thymikee) [inline __tests__/toMatchDiffSnapshot.test.js]: `test('works with default options', () => {})`

File: index.d.ts (hunk 3 above)
Comment (umidbekkarimov) [inline index.d.ts]: I'm not sure. I took [jest-immutable-matchers/index.d.ts](https://github.com/unindented/custom-immutable-matchers/blob/master/packages/jest-immutable-matchers/index.d.ts)  as an example cause it was working without any configuration with my IDE.

Comment (umidbekkarimov) [inline index.d.ts]: It's still in [docs](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html)

File: index.d.ts (hunk 2 above)
Comment (umidbekkarimov) [inline index.d.ts]: Done

File: __tests__/toMatchDiffSnapshot.test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    script
+`;
+
+beforeAll(() => {
+  expect.extend({ toMatchDiffSnapshot: snapshotDiff.toMatchDiffSnapshot });
+});
+
+test('works with default options', () => {
+  expect(a).toMatchDiffSnapshot(b);
+});
```
Comment (thymikee) [inline __tests__/toMatchDiffSnapshot.test.js]: Can you also make Flow happy? I'm fine with adding `// $FlowFixMe` here for now

**Comment by thymikee** [comment]
Thank you for your hard work @umidbekkarimov ❤️

**Pull request merged**